src/acquisition/covidcast/signal_dash_data_generator.py

```diff
@@ -184,25 +186,24 @@ def get_coverage(dashboard_signal: DashboardSignal,
     """Get the most recent coverage for the signal."""
     latest_time_value = get_latest_time_value_from_metadata(
         dashboard_signal, metadata)
+    start_day = latest_time_value - datetime.timedelta(days = LOOKBACK_DAYS_FOR_COVERAGE)
     latest_data = covidcast.signal(
         dashboard_signal.source,
         dashboard_signal.covidcast_signal,
-        end_day=latest_time_value,
-        start_day=latest_time_value)
+        end_day = latest_time_value,
+        start_day = start_day)
     count_by_geo_type_df = latest_data.groupby(
         ['geo_type', 'data_source', 'time_value', 'signal']).size().to_frame('count').reset_index()
 
-    if len(count_by_geo_type_df) > 1:
-        raise ValueError(f"Expected one row for coverage, got {len(count_by_geo_type_df)}.")
-
     signal_coverage_list = []
     
-    signal_coverage = DashboardSignalCoverage(
-        signal_id=dashboard_signal.db_id,
-        date=latest_time_value,
-        geo_type=count_by_geo_type_df['geo_type'].iloc[0],
-        count=count_by_geo_type_df['count'].iloc[0].item())
-    signal_coverage_list.append(signal_coverage)
+    for _, row in count_by_geo_type_df.iterrows():
+        signal_coverage = DashboardSignalCoverage(
+            signal_id=dashboard_signal.db_id,
+            date=row['time_value'].date(),
+            geo_type=row['geo_type'],
+            count=row['count'])
```

Comment:
What happens when we run this two days in a row:
* DashboardSignalCoverage timeseries for this signal (when retrieved using the signal dash API) gets longer by 1 day, though least-recent day was generated using (effectively) a different as_of than today's
* Previous DashboardSignalCoverage timeseries values for this timeseries are dropped first; timeseries remains 28 days long

Reply:
I think it's interesting to keep the older data -- would be neat to see the coverage going back many months. As you point out the caveat is that it may be out of date, but we could either disclaim that ("any changes that happen after 28 days may not be included") or add longer backfill occasionally (I'm not a huge fan of that since it adds complexity). I suspect it probably doesn't change too much.

If you prefer to keep it more conceptually simple for now we can just drop the older points, though.

Reply:
Cool, let's drop the older points then, it'll make the UI simpler/easier to interpret.

Reply:
Sounds good -- mind if I do this in a follow-up so we can get more data into the system for testing sooner?